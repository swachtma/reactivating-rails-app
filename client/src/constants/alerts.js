export const ADD_ALERT = 'ADD_ALERT'
export const CLEAR_ALERTS = 'CLEAR_ALERTS'

// Enumerated Values: default value should stay in array[0] of each list
export const scopes = ['application']
export const lvls = ['info', 'warning', 'success', 'danger']
