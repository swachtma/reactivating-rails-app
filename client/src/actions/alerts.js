import * as constants from "../constants/alerts";

export const addAlert = (message, level = constants.lvls[0], scope = constants.scopes[0]) => ({
  type: constants.ADD_ALERT, payload: {message, level, scope, timestamp: Date.now()}
});

export const clearAlerts = () => ({ type: constants.CLEAR_ALERTS }) ;