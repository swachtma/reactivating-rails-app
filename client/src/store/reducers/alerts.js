import * as constants from '../../constants/alerts'
const default_state = []

export const normalizeAlert = (payload) => {
  payload.scope = constants.scopes.includes(payload.scope) ? payload.scope : constants.scopes[0]
  payload.level = constants.lvls.includes(payload.level) ? payload.level : constants.lvls[0]
  return payload
}

export default function alertsReducer (state = default_state, action) {
  switch (action.type) {
    case (constants.ADD_ALERT):
      return [...state, normalizeAlert(action.payload)]
    case (constants.CLEAR_ALERTS):
      return []
    default:
      return state
  }
}
