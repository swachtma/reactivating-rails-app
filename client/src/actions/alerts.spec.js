/* global expect */
import * as actions from './alerts'

describe('Alerts actionCreators', () => {
  describe('addAlerts(message, level, scope)', () => {
    test('assigns defaults for scope and level', () => {
      let fsa = actions.addAlert('hello world')
      expect(fsa.payload.scope).toBeDefined()
      expect(fsa.payload.level).toBeDefined()
    })

    test('matches snapshot', () => {
      let snap = actions.addAlert('hello world')
      snap.payload.timestamp = 0
      expect(snap).toMatchSnapshot()
    })
  })

  describe('clearAlerts()', () => {
    test('matches snapshot', () => {
      expect(actions.clearAlerts()).toMatchSnapshot()
    })
  })
})
