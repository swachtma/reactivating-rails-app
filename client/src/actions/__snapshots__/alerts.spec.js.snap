// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts actionCreators addAlerts(message, level, scope) matches snapshot 1`] = `
Object {
  "payload": Object {
    "level": "info",
    "message": "hello world",
    "scope": "application",
    "timestamp": 0,
  },
  "type": "ADD_ALERT",
}
`;

exports[`Alerts actionCreators clearAlerts() matches snapshot 1`] = `
Object {
  "type": "CLEAR_ALERTS",
}
`;
